File: InsertionSort/C#/Insertion_sort.cs
using System.IO;
using System;

class Program
{
    static void Main()
    {
        int[] arr= {3,4,5,1,6,7,8,2,0};
        insertionSort(arr);
        
        foreach (int x in arr) 
        {
            Console.WriteLine(x);
        }
    }
    
    static void insertionSort(int[] arr)
    {
        for(int j=1; j < arr.Length; j++) 
        {
            int key = arr[j];
            int i = j - 1;
            while (i >= 0 && arr[i]>key)
            {
                arr[i+1] = arr[i];
                i = i - 1;
            }
            arr[i+1] = key;
        }
        
    }
}

File: Go/InsertionSort/InsertionSort.go
package main

import "fmt"

func InsertionSort(arr []int) {
	for i, val := range arr {
		j := i - 1
		for j >= 0 && arr[j] > val {
			arr[j+1] = arr[j]
			j = j - 1
		}
		arr[j + 1] = val
	}
}

func main() {
	arr := []int{2, 1, 1, 3}
	InsertionSort(arr)
	fmt.Println(arr)
}
